guidance/d_star_lite/d_star_lite/d_star_lite.py

```diff
@@ -0,0 +1,464 @@
+        self.goal = DSLNode(0, 0) # The goal DSLNode
+        self.U = [] # Priority queue
+        self.km = 0.0 # The minimum key in U
+        self.kold = 0.0 # The old minimum key in U
+        self.rhs = self.create_grid(float("inf")) # The right hand side values
+        self.g = self.create_grid(float("inf")) # The g values
+        self.initialized = False # Whether the grid has been initialized
+        self.WP = [] # The waypoints
+        self.dist_to_obstacle = dist_to_obstacle # The minimum distance a DSLNode must be from any obstacle to be considered valid
+
+    def create_grid(self, val: float) -> np.ndarray:
+        """
+        Creates a grid initialized with a specific value.
+
+        Args:
+            val (float): The value to initialize the grid with.
+
+        Returns:
+            np.ndarray: A 2D numpy array representing the initialized grid.
+        """
+        return np.full((self.x_max, self.y_max), val)
+    
+    def is_obstacle(self, DSLNode: DSLNode) -> bool:
+        """
+        Check if the DSLNode is considered an obstacle or is too close to an obstacle.
+
+        Args:
+            DSLNode (DSLNode): The DSLNode to check.
+
+        Returns:
+            bool: True if the DSLNode is too close to an obstacle or is an obstacle, False otherwise.
+        """
+        # Convert the DSLNode's coordinates to a numpy array for efficient distance computation
+        node_xy = np.array([DSLNode.x, DSLNode.y])
+
+        # Compute the euclidean distances from the DSLNode to all obstacles
+        distances = np.sqrt(np.sum((self.obstacles_xy - node_xy) ** 2, axis=1))
+
+        # Check if any distance is less than the minimum distance (default: 4.5)
+        return np.any(distances < self.dist_to_obstacle)
+    
+    def movement_cost(self, node1: DSLNode, node2: DSLNode) -> float:
+        """
+        Calculates the cost of moving from node1 to node2. Returns infinity if node2 is an obstacle or if no valid motion is found.
+
+        Args:
+            node1 (DSLNode): The starting DSLNode.
+            node2 (DSLNode): The ending DSLNode.
+
+        Returns:
+            float: The cost of moving from node1 to node2.
+        """
+        if self.is_obstacle(node2):
+            return math.inf
+        
+        movement_vector = DSLNode(node1.x - node2.x, node1.y - node2.y)
+        for motion in self.motions:
+            if compare_coordinates(motion, movement_vector):
+                return motion.cost
+        return math.inf
+    
+    def heuristic_distance(self, s: DSLNode) -> float:
+        """
+        Calculates the heuristic distance from DSLNode s to the goal using the Euclidean distance.
+
+        Args:
+            s (DSLNode): The DSLNode to calculate the heuristic distance from.
+
+        Returns:
+            float: The heuristic distance from DSLNode s to the goal.
+        """
+        return distance(s, self.goal) # Euclidean distance
```

Comment:
why not implement the distance function here instead of calling a non-member function